common/thorhelper/thorsoapcall.cpp

```diff
@@ -1291,6 +1296,30 @@ bool httpHeaderBlockContainsHeader(const char *httpheaders, const char *header)
         return true;
     return false;
 }
+
+bool getHTTPHeader(const char *httpheaders, const char *header, StringBuffer& value)
+{
+    if (!httpheaders || !*httpheaders || !header || !*header)
+        return false;
+
+    const char* pHeader = strstr(httpheaders, header);
+    if (!pHeader)
+        return false;
+
+    pHeader = strchr(pHeader, ':');
```

Comment:
You could still match on case "myheader-something-else:", how about:

pHeader+=strlen(header);
if (*pHeader!=':')
  return getHTTPHeader(pHeader, header, value); //not exact match, keep looking